### PR TITLE
SDIT-1561 Add nested field type to body part lists

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/model/translator.kt [hunk 1]
```diff
@@ -152,7 +152,7 @@ private fun IncentiveLevel?.toCurrentIncentive(): CurrentIncentive? = this?.let
   )
 }
 
-private fun List<BodyPartDetail>?.addIfContains(bodyPart: BodyPartDetail, keyword: String): List<BodyPartDetail>? =
+private fun List<BodyPartDetail>?.addIfCommentContains(bodyPart: BodyPartDetail, keyword: String): List<BodyPartDetail>? =
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/model/translator.kt:155]: Unrelated rename suggested in previous PR.

File: hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResource.kt [hunk 2]
```diff
@@ -48,6 +48,10 @@ class AttributeSearchResource(private val attributeSearchService: AttributeSearc
       Combining multiple sub-queries gives us the ability to create complex searches using any combination of a prisoner's 
       attributes. For example we can model nested queries such as <strong>"lastName IS Smith AND (prisonId IS MDI OR (prisonId IS OUT AND inOutStatus is OUT))"</strong>.
       </p>
+      <p>Please be aware that when searching lists of complex objects (e.g. aliases, alerts, tattoos) if you want to search for multiple attributes within the same object then you need 
+      to include them in the same query. For example, to search for alias "John Smith" you should search for aliases.firstName IS "John" and aliases.lastName IS "Smith" in a single query.
+      If you search for them in different queries you will receive anyone with firstName John and also anyone with lastName Smith.
+      </p>
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/resource/AttributeSearchResource.kt:54]: The main reason we have this single query rule is to avoid complexity in the code - otherwise in a complicated scenario with multiple queries and sub-queries and various ANDs / ORs it's really hard to know when to use a nested query for the attributes from nested objects. Just writing tests for it would be horrible. So we've compromised on this rule.

File: hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt [hunk 3]
```diff
@@ -74,6 +81,14 @@ class AttributeSearchIntegrationTest : AbstractSearchIntegrationTest() {
...
           BodyPartBuilder(bodyPart = "Shoulder", comment = "dragon"),
         ),
+        scar = listOf(
+          BodyPartBuilder(bodyPart = "Arm", comment = "wrist"),
+          BodyPartBuilder(bodyPart = "Shoulder", comment = "bite"),
+        ),
+        mark = listOf(
+          BodyPartBuilder(bodyPart = "Arm"),
+          BodyPartBuilder(bodyPart = "Shoulder", comment = "birthmark"),
+        ),
```
Comment (mikehalmamoj) [inline hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt:91]: The key to testing nested queries is that this prisoner shouldn't match e.g. tattooo Arm/dragon (and similar for scar Arm/bite and mark Arm/birthmark).

**Review by petergphillips — APPROVED** [review]
LGTM
